### PR TITLE
add isAuthenticating state

### PR DESCRIPTION
**Pull request by omonk**
I'm writing an app that uses Apollo and need to conditionally send users to different pages from the login callback based upon database responses. I found that when the app mounts it can take a little bit of time for all the authentication details to be processed, this `isAuthenticating` state allows me to use a lazy query and conditionally execute it when the time is ready.

I've tried to find all of the different areas where this `dispatch` needs to be fired but @Swizec  please check if I've missed anything else! 

### Sidenote: 

I feel the docs could be upgraded a bit as well, the guide is great but some details on the API of the `useAuth` would be worthwhile. Let me know your thoughts and I can have stab sometime this week,

### CODE REVIEW

**Comment by Swizec** [comment]
Thanks for the contribution @omonk this looks super useful. Merging :)

Good idea about the docs. I'll see what I can do.

File: README.md [hunk 1]
```diff
@@ -197,6 +228,8 @@ After logging out, Auth0 redirects back to your app. Again, it needs to know you
 
 ## Persisting login after refresh
 
+**NB Make sure you're not blocking blocking cookies! Extensions like privacy badger will prevent Auth0 from setting cookies so refreshing between logins wont work**
```
Comment (omonk) [inline README.md]: @Swizec realised I wrote blocking twice here!

Comment (Swizec) [reply README.md]: I'll fix it :)

**Comment by Swizec** [comment]
@all-contributors add @omonk for code

**Pull request merged**